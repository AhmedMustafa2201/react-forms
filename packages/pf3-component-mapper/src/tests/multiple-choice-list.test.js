import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import MockFieldProvider from '../../../../__mocks__/mock-field-provider';
import RequiredLabel from '../form-fields/required-label';
import { FieldLevelHelp, HelpBlock } from 'patternfly-react';
import Checkbox from '../files/checkbox';

describe.skip('<MultipleChoiceList />', () => {
  let initialProps;
  let changeSpy = jest.fn();
  beforeEach(() => {
    initialProps = {
      input: {
        name: 'Name of the field',
        value: ''
      },
      FieldProvider: (props) => <MockFieldProvider {...props} input={{ onChange: changeSpy, value: props.value || [] }} />,
      options: [
        {
          label: 'Foo',
          value: 0
        },
        {
          label: 'Bar',
          value: 1
        }
      ]
    };
  });

  afterEach(() => {
    changeSpy.mockReset();
  });

  it('should render correctly', () => {
    const wrapper = mount(<Checkbox {...initialProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call FieldProvider on change method', () => {
    const wrapper = mount(<Checkbox {...initialProps} />);

    wrapper
      .find('input')
      .last()
      .simulate('change', { target: { checked: true } });
    expect(changeSpy).toHaveBeenCalledWith([1]);
  });

  it('should call FieldProvider on change method and remove option value form all values', () => {
    const wrapper = mount(<Checkbox {...initialProps} value={[1]} />);

    wrapper
      .find('input')
      .last()
      .simulate('change', { target: { checked: true } });
    expect(changeSpy).toHaveBeenCalledWith([]);
  });

  it('should render in error state', () => {
    const wrapper = mount(
      <Checkbox
        {...initialProps}
        FieldProvider={(props) => (
          <MockFieldProvider
            {...props}
            input={{ onChange: changeSpy, value: [] }}
            meta={{
              error: 'Error message',
              touched: true
            }}
          />
        )}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render required variant', () => {
    const wrapper = mount(<Checkbox {...initialProps} isRequired label="Foo" />);
    expect(wrapper.find(RequiredLabel)).toHaveLength(1);
  });

  it('should render helper text variant', () => {
    const wrapper = mount(<Checkbox {...initialProps} isRequired label="Foo" helperText="Helper text" />);
    expect(wrapper.find(FieldLevelHelp)).toHaveLength(1);

    expect(
      wrapper
        .find(FieldLevelHelp)
        .first()
        .props().content
    ).toEqual('Helper text');
    expect(wrapper.find(HelpBlock)).toHaveLength(0);
  });

  it('should render description variant', () => {
    const wrapper = mount(<Checkbox {...initialProps} isRequired label="Foo" description="Description" />);
    expect(wrapper.find(FieldLevelHelp)).toHaveLength(0);
    expect(wrapper.find(HelpBlock)).toHaveLength(1);
    expect(
      wrapper
        .find(HelpBlock)
        .first()
        .text()
    ).toEqual('Description');
  });
});
