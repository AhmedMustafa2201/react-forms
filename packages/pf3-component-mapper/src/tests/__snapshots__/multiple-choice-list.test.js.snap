// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MultipleChoiceList /> should render correctly 1`] = `
<MultipleChoiceList
  FieldProvider={[Function]}
  options={
    Array [
      Object {
        "label": "Foo",
        "value": 0,
      },
      Object {
        "label": "Bar",
        "value": 1,
      },
    ]
  }
>
  <MultipleChoiceList
    Checkbox={[Function]}
    FieldProvider={[Function]}
    Wrapper={[Function]}
    options={
      Array [
        Object {
          "label": "Foo",
          "value": 0,
        },
        Object {
          "label": "Bar",
          "value": 1,
        },
      ]
    }
  >
    <FieldProvider
      options={
        Array [
          Object {
            "label": "Foo",
            "value": 0,
          },
          Object {
            "label": "Bar",
            "value": 1,
          },
        ]
      }
      render={[Function]}
      validate={[Function]}
    >
      <MockFieldProvider
        input={
          Object {
            "onChange": [MockFunction],
            "value": Array [],
          }
        }
        options={
          Array [
            Object {
              "label": "Foo",
              "value": 0,
            },
            Object {
              "label": "Bar",
              "value": 1,
            },
          ]
        }
        render={[Function]}
        validate={[Function]}
      >
        <Wrapper
          meta={Object {}}
          rest={
            Object {
              "children": undefined,
              "input": Object {
                "onChange": [MockFunction],
                "value": Array [],
              },
              "validate": [Function],
            }
          }
        >
          <FormGroup
            bsClass="form-group"
            validationState={null}
          >
            <div
              className="form-group"
            >
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                <label
                  className="control-label"
                />
              </ControlLabel>
              <div>
                <FieldProvider
                  id="undefined-0"
                  key="0"
                  label="Foo"
                  render={[Function]}
                  type="checkbox"
                  value={0}
                >
                  <MockFieldProvider
                    id="undefined-0"
                    input={
                      Object {
                        "onChange": [MockFunction],
                        "value": Array [],
                      }
                    }
                    label="Foo"
                    render={[Function]}
                    type="checkbox"
                    value={0}
                  >
                    <FinalCheckbox
                      aria-label="Foo"
                      id="undefined-0"
                      label="Foo"
                      onChange={[Function]}
                      type="checkbox"
                      value={0}
                    >
                      <Checkbox
                        aria-label="Foo"
                        bsClass="checkbox"
                        disabled={false}
                        id="undefined-0"
                        inline={false}
                        onChange={[Function]}
                        title=""
                        type="checkbox"
                        value={0}
                      >
                        <div
                          className="checkbox"
                        >
                          <label
                            title=""
                          >
                            <input
                              aria-label="Foo"
                              disabled={false}
                              id="undefined-0"
                              onChange={[Function]}
                              type="checkbox"
                              value={0}
                            />
                            Foo
                          </label>
                        </div>
                      </Checkbox>
                    </FinalCheckbox>
                  </MockFieldProvider>
                </FieldProvider>
                <FieldProvider
                  id="undefined-1"
                  key="1"
                  label="Bar"
                  render={[Function]}
                  type="checkbox"
                  value={1}
                >
                  <MockFieldProvider
                    id="undefined-1"
                    input={
                      Object {
                        "onChange": [MockFunction],
                        "value": 1,
                      }
                    }
                    label="Bar"
                    render={[Function]}
                    type="checkbox"
                    value={1}
                  >
                    <FinalCheckbox
                      aria-label="Bar"
                      id="undefined-1"
                      label="Bar"
                      onChange={[Function]}
                      type="checkbox"
                      value={1}
                    >
                      <Checkbox
                        aria-label="Bar"
                        bsClass="checkbox"
                        disabled={false}
                        id="undefined-1"
                        inline={false}
                        onChange={[Function]}
                        title=""
                        type="checkbox"
                        value={1}
                      >
                        <div
                          className="checkbox"
                        >
                          <label
                            title=""
                          >
                            <input
                              aria-label="Bar"
                              disabled={false}
                              id="undefined-1"
                              onChange={[Function]}
                              type="checkbox"
                              value={1}
                            />
                            Bar
                          </label>
                        </div>
                      </Checkbox>
                    </FinalCheckbox>
                  </MockFieldProvider>
                </FieldProvider>
              </div>
            </div>
          </FormGroup>
        </Wrapper>
      </MockFieldProvider>
    </FieldProvider>
  </MultipleChoiceList>
</MultipleChoiceList>
`;

exports[`<MultipleChoiceList /> should render in error state 1`] = `
<MultipleChoiceList
  FieldProvider={[Function]}
  options={
    Array [
      Object {
        "label": "Foo",
        "value": 0,
      },
      Object {
        "label": "Bar",
        "value": 1,
      },
    ]
  }
>
  <MultipleChoiceList
    Checkbox={[Function]}
    FieldProvider={[Function]}
    Wrapper={[Function]}
    options={
      Array [
        Object {
          "label": "Foo",
          "value": 0,
        },
        Object {
          "label": "Bar",
          "value": 1,
        },
      ]
    }
  >
    <FieldProvider
      options={
        Array [
          Object {
            "label": "Foo",
            "value": 0,
          },
          Object {
            "label": "Bar",
            "value": 1,
          },
        ]
      }
      render={[Function]}
      validate={[Function]}
    >
      <MockFieldProvider
        input={
          Object {
            "onChange": [MockFunction],
            "value": Array [],
          }
        }
        meta={
          Object {
            "error": "Error message",
            "touched": true,
          }
        }
        options={
          Array [
            Object {
              "label": "Foo",
              "value": 0,
            },
            Object {
              "label": "Bar",
              "value": 1,
            },
          ]
        }
        render={[Function]}
        validate={[Function]}
      >
        <Wrapper
          error="Error message"
          meta={
            Object {
              "error": "Error message",
              "touched": true,
            }
          }
          rest={
            Object {
              "children": undefined,
              "input": Object {
                "onChange": [MockFunction],
                "value": Array [],
              },
              "validate": [Function],
            }
          }
          showError="Error message"
        >
          <FormGroup
            bsClass="form-group"
            validationState="error"
          >
            <div
              className="form-group has-error"
            >
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                <label
                  className="control-label"
                />
              </ControlLabel>
              <div>
                <FieldProvider
                  id="undefined-0"
                  key="0"
                  label="Foo"
                  render={[Function]}
                  type="checkbox"
                  value={0}
                >
                  <MockFieldProvider
                    id="undefined-0"
                    input={
                      Object {
                        "onChange": [MockFunction],
                        "value": Array [],
                      }
                    }
                    label="Foo"
                    meta={
                      Object {
                        "error": "Error message",
                        "touched": true,
                      }
                    }
                    render={[Function]}
                    type="checkbox"
                    value={0}
                  >
                    <FinalCheckbox
                      aria-label="Foo"
                      id="undefined-0"
                      label="Foo"
                      onChange={[Function]}
                      type="checkbox"
                      value={0}
                    >
                      <Checkbox
                        aria-label="Foo"
                        bsClass="checkbox"
                        disabled={false}
                        id="undefined-0"
                        inline={false}
                        onChange={[Function]}
                        title=""
                        type="checkbox"
                        value={0}
                      >
                        <div
                          className="checkbox"
                        >
                          <label
                            title=""
                          >
                            <input
                              aria-label="Foo"
                              disabled={false}
                              id="undefined-0"
                              onChange={[Function]}
                              type="checkbox"
                              value={0}
                            />
                            Foo
                          </label>
                        </div>
                      </Checkbox>
                    </FinalCheckbox>
                  </MockFieldProvider>
                </FieldProvider>
                <FieldProvider
                  id="undefined-1"
                  key="1"
                  label="Bar"
                  render={[Function]}
                  type="checkbox"
                  value={1}
                >
                  <MockFieldProvider
                    id="undefined-1"
                    input={
                      Object {
                        "onChange": [MockFunction],
                        "value": Array [],
                      }
                    }
                    label="Bar"
                    meta={
                      Object {
                        "error": "Error message",
                        "touched": true,
                      }
                    }
                    render={[Function]}
                    type="checkbox"
                    value={1}
                  >
                    <FinalCheckbox
                      aria-label="Bar"
                      id="undefined-1"
                      label="Bar"
                      onChange={[Function]}
                      type="checkbox"
                      value={1}
                    >
                      <Checkbox
                        aria-label="Bar"
                        bsClass="checkbox"
                        disabled={false}
                        id="undefined-1"
                        inline={false}
                        onChange={[Function]}
                        title=""
                        type="checkbox"
                        value={1}
                      >
                        <div
                          className="checkbox"
                        >
                          <label
                            title=""
                          >
                            <input
                              aria-label="Bar"
                              disabled={false}
                              id="undefined-1"
                              onChange={[Function]}
                              type="checkbox"
                              value={1}
                            />
                            Bar
                          </label>
                        </div>
                      </Checkbox>
                    </FinalCheckbox>
                  </MockFieldProvider>
                </FieldProvider>
              </div>
              <HelpBlock
                bsClass="help-block"
              >
                <span
                  className="help-block"
                >
                  Error message
                </span>
              </HelpBlock>
            </div>
          </FormGroup>
        </Wrapper>
      </MockFieldProvider>
    </FieldProvider>
  </MultipleChoiceList>
</MultipleChoiceList>
`;
